### ISSUE TITLE
Question regarding MaxIssuesToPost

### ISSUE THREAD

**Issue by janniksam** [cake-contrib/Cake.Issues.PullRequests.AzureDevOps#165]
Lets imagine, in theory, my pullrequest is fixing somehing in a old legacy classes and 200 warnings pop up during code analyse of my PRs build. Most of aren't results from by my changes. MaxIssuesToPost is set to 50 warnings, so 50 warnings will be output to the Pull Request..

Now I'll fix all the 50 warnings in my next commit to the PR. The build will be rerun and reanalyse everything. **Will it output the next 50 warnings or does it detect that it already detect the 50 submitted warnings?**

It already gave me 50, and I wouldn't want to fix another 50, when it comes to legacy code. We're aiming for small but steady steps to our zero warning policy. (currently we're at 14000 ;-))

**Comment by pascalberger** [comment]
> Will it output the next 50 warnings or does it detect that it already detect the 50 submitted warnings?

Yes, in this case it will mark the first 50 as resolved and post the next 50.

This is how I currently approach this scenario:
1. We don't mix fixing of existing code analysis issues with other PRs. So if we create a PR changing something in an existing code base, we normally won't fix the issues in the same PR. If you have a policy that comments need to be resolved you can mark the comments as `Won't Fix` (if you mark them as `Resolved` they will be reopened the next run by the addin since they are not actually resolved)
2. We added logic to our build script, which filters out existing issues and only posts issues which are introduced by this PR to the build script. This is done by the following steps:
  1. Persist issues for every build and store them as build artifacts
  2. Read target branch from PR
  3. Detect common Git commit for source and target branch
  4. Detect build for the common Git commit
  5. Read issues from build artifacts
  6. Deserialize issues
  7. Filter existing issues from the list of issues posted to the pull request

For most of the steps above there are aliases and helpers available, but there's  currently no out-of-the box implementation for this. It might be possible to add this to Cake.Issues.Recipe, but I'm not sure how easy it is to implement this in a generic way. But if you're interested in this approach I definitely help you with code snippets.

Another possibility would be to introduce a new setting which limits number of issues posted to a single pull request across multiple build runs. This should be possible, since we have marked issues posted by the addin and are already reading them back.

Let me know if any of these approaches help in your use case.

**Comment by janniksam** [comment]
> Another possibility would be to introduce a new setting which limits number of issues posted to a single pull request across multiple build runs. This should be possible, since we have marked issues posted by the addin and are already reading them back.

Thanks for always getting back to me so quickly. Thats really helpful.

Unfortunately I don't have the spare time to work at that first time-connsuming algorithm you wrote down for me (which I've understood completely) . Maybe in the future, but I am already way past my time estimate implementing the code analysis for my company.

The approach, that I had personally in mind, you actually thought about aswell:

"Another possibility would be to introduce a new setting which limits number of issues posted to a single pull request across multiple build runs. This should be possible, since we have marked issues posted by the addin and are already reading them back."

That seems to be a very quick (subjective statement :-)) but ok'ish way to do the "capping". I figured, you could detect the already posted comments. I know that it does have some negative aspects (e.g. new warnings could be skipped, when 100+ already existing warnings are priorized), but I could live with that for now.

**I would love to see that, if it doesnt take too much time.** If I can help you somehow, please let me know.
